src/Domain/RegularGivingService.php

```diff
@@ -212,8 +214,6 @@ public function setupNewMandate(
             );
```

Comment:
the throw just above should now never happen - if the donation was not collected presumably `PaymentIntentNotSucceeded` was thrown at line 188 above, so I don't think the if check here is useful.

Reply:
I think we've been discussing two alternative changes around that exception which are mutually exclusive, so I'm not really clear on the best change here now, and don't want to apply one unilaterally. If we narrow the exception's meaning to be _failure_ statuses (`canceled`) and not pending ones, then I think this remains needed.

Reply:
Yeah, compared to always catching there is no difference. I don't feel strongly that we need the exception.